config/initializers/devise.rb

```diff
@@ -142,3 +142,25 @@
   # Defines name of the authentication token params key
   config.token_authentication_key = :auth_token
 end
+
+if ENV["OPENID_APP_ID"].present? && ENV["OPENID_APP_SECRET"].present?
+  Devise.setup do |config|
+    protocol = Rails.env.development? ? "http://" : "https://"
+    config.omniauth :openid_connect, {
+      name: :openid_connect,
+      issuer: "https://login.lescommuns.org/auth/realms/data-food-consortium",
+      scope: [:openid, :profile, :email],
+      response_type: :code,
+      uid_field: "email",
+      discovery: true,
+      client_auth_method: :jwks,
+
+      client_options: {
+        identifier: ENV["OPENID_APP_ID"],
+        secret: ENV["OPENID_APP_SECRET"],
+        redirect_uri: "#{protocol}#{ActionMailer::Base.default_url_options[:host]}/user/spree_user/auth/openid_connect/callback"
```

Comment:
Is this line missing a comma? Latest pull fails to initialise. 
```suggestion
        redirect_uri: "#{protocol}#{ActionMailer::Base.default_url_options[:host]}/user/spree_user/auth/openid_connect/callback",
```

Reply:
Yes, looks like you are right. Andrew stopped work on this though. @Matt-Yorkley are you taking over?

Reply:
I guess so...